tmva/sofie/inc/TMVA/SOFIEHelpers.hxx

```diff
@@ -0,0 +1,60 @@
+#ifndef TMVA_SOFIE_SOFIE_HELPERS
+#define TMVA_SOFIE_SOFIE_HELPERS
+
+
+#include <type_traits>
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include <vector>
+#include <memory>
+#include <regex>
+
+namespace TMVA{
+namespace Experimental{
+
+// Functor to wrap a SOFIE session to RDF functor signature
+
+template <typename I, typename F, typename T>
+class SofieFunctorHelper;
+
+template <std::size_t... N,  typename S, typename T>
+class SofieFunctorHelper<std::index_sequence<N...>, S, T> {
+   /// this is the magic to defined the operator () with N fixed parameter arguments
+   template <std::size_t Idx>
+   using AlwaysT = T;
+
+   std::vector<std::vector<T>> fInput;
+   std::vector<std::shared_ptr<S>> fSessions;
```

Comment:
I prefer using shared pointer since I don't want to prevent the case that one can make copies of SofieFunctorHelper's

Reply:
`std::vector<S>` would not prevent copies and it would have more predictable behavior in case of copies:

```cpp
auto helper2 = helper1;
// currently using helper2 and helper1 concurrently is thread-unsafe
// because they share the underlying session. this might be counter-intuitive
```